### PR TITLE
[bitnami/mongodb-sharded] MongoDB sharded standarization

### PR DESCRIPTION
**Pull request by fmulero**
Signed-off-by: Fran Mulero <fmulero@vmware.com>

**Description of the change**

This PR updates mongodb to its major version 5.0 and applies standardisation changes to be in line with the rest of the catalog.

**Checklist**

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the values.yaml and added to the README.md using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Review by migruiz4 — CHANGES_REQUESTED** [review]
The following values are missing in `values.yaml`:
```
mongodbDisableSystemLog
auth.existingAdminSecret
```

The following values weren't implemented (may be false positives):
```
common.podLabels
common.serviceAccount.annotations
common.serviceAccount.automountServiceAccountToken
service.externalTrafficPolicy
configsvr.persistence.subPath
configsvr.persistence.storageClass
mongos.servicePerReplica.externalTrafficPolicy
mongos.servicePerReplica.externalIPs
mongos.serviceAccount.create
mongos.serviceAccount.name
mongos.serviceAccount.annotations
mongos.serviceAccount.automountServiceAccountToken
shardsvr.dataNode.updateStrategy.type
shardsvr.dataNode.serviceAccount.create
shardsvr.dataNode.serviceAccount.name
shardsvr.dataNode.serviceAccount.annotations
shardsvr.dataNode.serviceAccount.automountServiceAccountToken
shardsvr.dataNode.containerSecurityContext.runAsNonRoot
shardsvr.dataNode.containerSecurityContext.readOnlyRootFilesystem
shardsvr.dataNode.livenessProbe.initialDelaySeconds
shardsvr.dataNode.livenessProbe.periodSeconds
shardsvr.dataNode.livenessProbe.timeoutSeconds
shardsvr.dataNode.livenessProbe.failureThreshold
shardsvr.dataNode.livenessProbe.successThreshold
shardsvr.dataNode.readinessProbe.initialDelaySeconds
shardsvr.dataNode.readinessProbe.periodSeconds
shardsvr.dataNode.readinessProbe.timeoutSeconds
shardsvr.dataNode.readinessProbe.failureThreshold
shardsvr.dataNode.readinessProbe.successThreshold
shardsvr.dataNode.startupProbe.initialDelaySeconds
shardsvr.dataNode.startupProbe.periodSeconds
shardsvr.dataNode.startupProbe.timeoutSeconds
shardsvr.dataNode.startupProbe.failureThreshold
shardsvr.dataNode.startupProbe.successThreshold
shardsvr.persistence.subPath
shardsvr.persistence.storageClass
shardsvr.arbiter.updateStrategy.type
shardsvr.arbiter.serviceAccount.create
shardsvr.arbiter.serviceAccount.name
shardsvr.arbiter.serviceAccount.annotations
shardsvr.arbiter.serviceAccount.automountServiceAccountToken
shardsvr.arbiter.podSecurityContext.fsGroup
shardsvr.arbiter.containerSecurityContext.runAsUser
shardsvr.arbiter.containerSecurityContext.runAsNonRoot
shardsvr.arbiter.containerSecurityContext.readOnlyRootFilesystem
shardsvr.arbiter.livenessProbe.initialDelaySeconds
shardsvr.arbiter.livenessProbe.periodSeconds
shardsvr.arbiter.livenessProbe.timeoutSeconds
shardsvr.arbiter.livenessProbe.failureThreshold
shardsvr.arbiter.livenessProbe.successThreshold
shardsvr.arbiter.readinessProbe.initialDelaySeconds
shardsvr.arbiter.readinessProbe.periodSeconds
shardsvr.arbiter.readinessProbe.timeoutSeconds
shardsvr.arbiter.readinessProbe.failureThreshold
shardsvr.arbiter.readinessProbe.successThreshold
shardsvr.arbiter.startupProbe.initialDelaySeconds
shardsvr.arbiter.startupProbe.periodSeconds
shardsvr.arbiter.startupProbe.timeoutSeconds
shardsvr.arbiter.startupProbe.failureThreshold
shardsvr.arbiter.startupProbe.successThreshold
metrics.podSecurityContext.enabled
metrics.podSecurityContext.fsGroup
```

**Comment by fmulero** [comment]
@migruiz4 Could you review it again? Most of the values you pointed out before are false positives.

**Comment by migruiz4** [comment]
Thanks @fmulero, I fixed most of the false positives but these values are still missing:

```
mongos.servicePerReplica.externalTrafficPolicy
mongos.servicePerReplica.externalIPs
metrics.podSecurityContext.enabled
metrics.podSecurityContext.fsGroup
```

File: bitnami/mongodb-sharded/README.md [hunk 1]
```diff
@@ -487,13 +637,34 @@ Find more information about how to deal with common errors related to Bitnami's
...
+### To 5.0.0
+
+This major release renames several values in this chart and adds missing features, in order to be inline with the rest of assets in the Bitnami charts repository.
+
+Affected values:
+
+- Authentication parameters are reorganized under the `auth.*` parameter:
+  - `usePassword` is renamed to `auth.enabled`.
+  - `mongodbRootPassword`, `mongodbUsername`, `mongodbPassword`, `mongodbDatabase`, and `replicaSetKey` are now `auth.rootPassword`, `auth.username`, `auth.password`, `auth.database`, and `auth.replicaSetKey` respectively.
+- `common.containerPorts.mongo` is renamed to `common.containerPorts.mongodb`
```
Comment (migruiz4) [inline bitnami/mongodb-sharded/README.md:657]: Missing references to `existingSecret` and `usePasswordFile`

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Just a comment in addition to my previous request

**Comment by fmulero** [comment]
Thanks @migruiz4  for your findings. These parameter were removed, they are not needed (metrics container is a sidecar):
```
metrics.podSecurityContext.enabled
metrics.podSecurityContext.fsGroup
```

About `servicePerReplica` values, I rewrote the `mongos-service-per-replica.yaml` template, Could you take it a look?

**Review by migruiz4 — DISMISSED** [review]
LGTM!

**Pull request merged**